spec/ReadPreferenceOption.spec.js

```diff
@@ -0,0 +1,978 @@
+'use strict'
+
+const Parse = require('parse/node');
+const ReadPreference = require('mongodb').ReadPreference;
+const rp = require('request-promise');
+
+describe('Read preference option', () => {
+  it('should find in primary by default', (done) => {
+    const obj0 = new Parse.Object('MyObject');
+    obj0.set('boolKey', false);
+    const obj1 = new Parse.Object('MyObject');
+    obj1.set('boolKey', true);
+
+    Parse.Object.saveAll([obj0, obj1]).then(() => {
+      const hook = {
+        method: function(preference) {
+          expect(preference).toEqual(ReadPreference.PRIMARY);
+        }
+      };
+      spyOn(hook, 'method').and.callThrough();
+
+      const cursor = databaseAdapter.database.serverConfig.cursor;
```

Comment:
the databaseAdapter should probably be acquired by `new Config('test').databaseController.adapter`